crates/starknet-devnet-core/src/starknet/mod.rs

```diff
@@ -106,6 +107,7 @@ pub struct Starknet {
     pub config: StarknetConfig,
     pub pending_block_timestamp_shift: i64,
     pub next_block_timestamp: Option<u64>,
+    pub next_block_gas_update: Option<GasUpdate>,
```

Comment:
I implemented it the same way as  next_block_timestamp based on this comment https://github.com/0xSpaceShard/starknet-devnet-rs/pull/551#discussion_r1679524258. 

It's `None`, by default and if it's set we use it.

But of course I can change it to have default gas and always use it (remove option), maybe it will simplify code related to get data.

Reply:
refactored in 1627c07726becc7fdddac2c50973f53eca7dba53